File: auth0/src/test/java/com/auth0/android/management/ManagementExceptionTest.java
package com.auth0.android.management;

import com.auth0.android.NetworkErrorException;

import org.junit.Test;

import java.io.IOException;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class ManagementExceptionTest {

    @Test
    public void shouldNotHaveNetworkError() {
        ManagementException ex = new ManagementException("Something else happened");
        assertThat(ex.isNetworkError(), is(false));
    }

    @Test
    public void shouldHaveNetworkError() {
        ManagementException ex = new ManagementException("Request has definitely failed", new NetworkErrorException(new IOException()));
        assertThat(ex.isNetworkError(), is(true));
    }
}

File: auth0/src/test/java/com/auth0/android/management/ManagementExceptionTest.kt
package com.auth0.android.management

import com.auth0.android.NetworkErrorException
import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.junit.Test
import java.io.IOException

public class ManagementExceptionTest {
    @Test
    public fun shouldNotHaveNetworkError() {
        val ex = ManagementException("Something else happened")
        MatcherAssert.assertThat(ex.isNetworkError, CoreMatchers.`is`(false))
    }

    @Test
    public fun shouldHaveNetworkError() {
        val ex = ManagementException(
            "Request has definitely failed", NetworkErrorException(
                IOException()
            )
        )
        MatcherAssert.assertThat(ex.isNetworkError, CoreMatchers.`is`(true))
    }
}